projects/components/src/load-async/load-async.directive.ts

```diff
@@ -65,6 +60,23 @@ export class LoadAsyncDirective implements OnChanges, OnDestroy {
   private buildWrapperView(): ComponentRef<LoadAsyncWrapperComponent> {
     const componentFactory = this.componentFactoryResolver.resolveComponentFactory(LoadAsyncWrapperComponent);
 
+    // Second param for structural directive is undefined until this.data$ is defined
+    // So putting this assignment in constuctor will not work
+    // This will execute only once as this method is called only
+    this.wrapperInjector = Injector.create({
```

Comment:
We don't need to create a new injector. It would be problematic and also it wouldn't work since right now we are only creating a new instance of component when data$ reference itself change. New emits on data$ is automatically passed via the state$ property.

What I would recommend it to modify line 45 instead and add loaderType$ as a property in `LoadAsyncWrapperParameters`. 

Create a `ReplaySubject` for loaderType -> 
```
private readonly loaderTypeSubject: BehaviourSubject<LoaderType> = new BehaviourSubject(<default-value>);
private readonly loaderType$: Observable<LoaderType>= this.loaderTypeSubject.asObservable();
```

Then use add loaderType$ on like 46. 

and add a separate ngOnChanges trigger for loaderType, that would just emit on this subject.

Reply:
updated